lib/phoenix_live_view.ex

```diff
@@ -758,4 +814,32 @@ defmodule Phoenix.LiveView do
   defmacro sigil_L({:<<>>, _, [expr]}, []) do
     EEx.compile_string(expr, engine: Phoenix.LiveView.Engine, line: __CALLER__.line + 1)
   end
+
+  @doc """
+  Generates a live link for HTML5 pushState based navigation without page reloads.
+
+  ## Options
+
+    * `:to` - the required path to link to.
+    * `:replace` - the flag to replace the current history or push a new state.
+      Defaults `false`.
+
+  ## Examples
+
+      <%= live_link "next", to: Routes.live_path(@socket, MyLive, @page + 1) %>
+      <%= live_link to: Routes.live_path(@socket, MyLive, dir: :asc), replace: false do %>
+        Sort By Price
+      <% end %>
+
+  """
+  def live_link(opts) when is_list(opts), do: live_link(opts, do: Keyword.fetch!(opts, :do))
+  def live_link(opts, do: block) when is_list(opts) do
+    uri = Keyword.fetch!(opts, :to)
+    replace = Keyword.get(opts, :replace, false)
+    kind = if replace, do: "replace", else: "push"
+    Phoenix.HTML.Link.link(to: uri, data: [phx_live_link: kind], do: block)
```

Comment:
I think calling `content_tag(:a, href: uri, data: [phx_live_link: kind], do: block)` may work better here as we don't need all idiosyncrasies from link.